doc/nrf/ug_thread.rst

```diff
@@ -0,0 +1,183 @@
+* ``CONFIG_OPENTHREAD_COAP`` - This option enables OpenThread's native CoAP API.
+
+You can also change the default values for the following options:
+
+* ``CONFIG_OPENTHREAD_CHANNEL`` - By default set to ``11``.
+  You can set any value ranging from ``11`` to ``26``.
+* ``CONFIG_OPENTHREAD_PANID`` - By default set to ``43981``.
+  You can set any value ranging from ``0`` to ``65535``.
+
+Thread commissioner
+-------------------
+
+Thread commissioner coordinates the process of adding new Thread devices to the network.
+It is optional, because the Thread :ref:`samples` in |NCS| use a hardcoded network information.
+
+To enable Thread commissioner, set the following Kconfig options to the provided values:
+
+* ``CONFIG_OPENTHREAD_COMMISSIONER`` to ``y``.
+* ``CONFIG_MBEDTLS_HEAP_SIZE`` to ``8192``.
+* ``CONFIG_SHELL_STACK_SIZE`` to ``3072``.
+
+OpenThread stack logging options
+--------------------------------
+
+The OpenThread stack logging is handled with the following options:
+
+* ``CONFIG_LOG`` - This option enables Zephyr's :ref:`zephyr:logger`.
+* ``CONFIG_OPENTHREAD_DEBUG`` - This option enables logging for the OpenThread stack.
+
+Both options must be enabled to allow logging.
+
+This said, enabling logging is optional, because it is enabled by default for all Thread samples.
+However, you must set one of the following logging levels to start receiving the logging output:
+
+* ``CONFIG_OPENTHREAD_LOG_LEVEL_ERROR`` - Enables logging only for errors.
+* ``CONFIG_OPENTHREAD_LOG_LEVEL_WARNING`` - Enables logging for errors and warnings.
+* ``CONFIG_OPENTHREAD_LOG_LEVEL_INFO`` - Enables logging for informational messages, errors, and warnings.
+* ``CONFIG_OPENTHREAD_LOG_LEVEL_DEBUG`` - Enables logging for debug messages, informational messages, errors, and warnings.
+
+Zephyr L2 logging options
+-------------------------
+
+If you want to get logging output related to the Zephyr's L2 layer, enable one of the following Kconfig options:
+
+* ``CONFIG_OPENTHREAD_L2_LOG_LEVEL_ERR`` - Enables logging only for errors.
+* ``CONFIG_OPENTHREAD_L2_LOG_LEVEL_WRN`` - Enables logging for errors and warnings.
+* ``CONFIG_OPENTHREAD_L2_LOG_LEVEL_INF`` - Enables logging for informational messages, errors, and warnings.
+* ``CONFIG_OPENTHREAD_L2_LOG_LEVEL_DBG`` - Enables logging for debug messages, informational messages, errors, and warnings.
+
+Chosing one of these options will enable writing the appropriate information in the L2 debug log.
+
+Additionally, enabling ``CONFIG_OPENTHREAD_L2_LOG_LEVEL_DBG`` allows you to set the ``CONFIG_OPENTHREAD_L2_DEBUG`` option, which in turn has the following settings:
+
+* ``CONFIG_OPENTHREAD_L2_DEBUG_DUMP_15_4``
+* ``CONFIG_OPENTHREAD_L2_DEBUG_DUMP_IPV6``
+
+These options enable dumping 802.15.4 or IPv6 frames (or both) in the debug log output.
+
+You can disable writing to log with the ``CONFIG_OPENTHREAD_L2_LOG_LEVEL_OFF`` option.
+
```

Comment:
Shall we also mention how can one switch between device types? Here are the available options:
https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/net/l2/openthread/Kconfig#L133
By default Zephyr will use FTD libraries, but with those configs MTD can be used. With MTD enabled it's possible to configure SED.